### PR TITLE
Enhance grep tool: add context, ignoreCase, literal, limit parameters

### PR DESCRIPTION
**Pull request by tiancaiamao**
## Summary

Enhance the grep tool with 4 new optional parameters to reduce agent fallback to bash grep commands:

### New Parameters
| Parameter | Type | Default | Description |
|-----------|------|---------|-------------|
| `ignoreCase` | bool | false | Case-insensitive search (`-i`) |
| `literal` | bool | false | Treat pattern as literal string (`-F`) |
| `context` | int | 0 | Context lines before/after match (`-C`) |
| `limit` | int | 100 | Max matches returned |

### Changes
- **`pkg/tools/grep.go`**: Added parameters, updated JSON schema, updated rg/grep command building, added output limiting
- **`pkg/tools/grep_test.go`**: New test file with 10 test cases covering all new parameters
- **`pkg/prompt/prompt.md`**: Updated tool selection guidance

### Backward Compatibility
All new parameters are optional. Existing calls without them behave identically to before.

### Test Results
```
=== RUN   TestGrepTool_NameAndDescription      --- PASS
=== RUN   TestGrepTool_BasicSearch              --- PASS
=== RUN   TestGrepTool_IgnoreCase               --- PASS
=== RUN   TestGrepTool_Literal                  --- PASS
=== RUN   TestGrepTool_ContextLines             --- PASS
=== RUN   TestGrepTool_Limit                    --- PASS
=== RUN   TestGrepTool_FilePattern              --- PASS
=== RUN   TestGrepTool_NoMatches                --- PASS
=== RUN   TestGrepTool_PathExpansion            --- PASS
=== RUN   TestGrepTool_AllParametersCombined    --- PASS
PASS
```

### CODE REVIEW

**Comment by tiancaiamao** [comment]
🔍 **Self-Review for PR #189**

## Summary of Changes
- Enhanced `pkg/tools/grep.go` with 4 new optional parameters: `ignoreCase`, `literal`, `context`, `limit`
- Added 10 test cases in `pkg/tools/grep_test.go` — all passing
- Updated tool guidance in `pkg/prompt/prompt.md`

## Review Findings

### No P0 issues found

### P1 — Minor optimization opportunity (not blocking)
1. **Limit enforcement is post-processing**: The `limitMatches()` function truncates output after rg/grep has already produced all results. For very large result sets, passing `-m <limit>` to rg would be more efficient. However, this is an optimization — the current approach is correct and the default limit of 100 keeps output manageable. Can be addressed in a follow-up if needed.

2. **`isContextLine()` heuristic**: The function uses a pattern-based heuristic to distinguish match lines (`file:line:text`) from context lines (`file-line-text`). Edge cases with filenames containing `-<digits>-` patterns could theoretically be misidentified, but this is unlikely in practice and the heuristic works correctly for the common case.

### P2 — Cosmetic
- Test file missing trailing newline (minor)

## Test Results
All 10 grep tests + full `pkg/tools` suite pass.

## Verdict
✅ **Review passed.** No P0/P1 blocking issues. Implementation is clean, backward-compatible, and well-tested.

File: pkg/prompt/prompt.md [hunk 1]
```diff
@@ -51,8 +51,7 @@ For detailed guidelines and templates, check available skills.
 
 **For investigation/debugging tasks:**
 - Use `grep` first to locate relevant code (search for error messages, function names, patterns)
-- Only read files after grep identifies relevant locations
-- Avoid reading entire codebases blindly — target your investigation
```
Comment (tiancaiamao) [inline pkg/prompt/prompt.md:55]: The old lines are still useful? They try to avoid LLM using read offset + limit as paging method and repeat a lot tool calls...

**Comment by tiancaiamao** [comment]
🔍 **Self-Review Round 2 (post address-comment)**

## Changes in address-comment commit (3568143)

### P1 Fix: Limit enforcement
- Added `-m <limit>` flag to rg command for early termination instead of post-processing only
- grep fallback relies on `limitMatches()` post-processing (grep `-m` counts context lines too, making it unreliable)
- `limitMatches()` kept as safety net for rg as well

### P1 Fix: isContextLine() heuristic
- Replaced left-to-right first-match approach with rightmost-separator strategy
- The line number field is always the rightmost `:<digits>:` or `-<digits>-` pattern
- Now correctly handles filenames like `report-2024-01.txt:5:content` (match line, not context)
- Added `TestIsContextLine` with 9 test cases including edge cases

### P2 Fix: Trailing newline added to test file

## Test Results
All 12 grep tests + full `pkg/tools` suite pass.

## Verdict
✅ **Review passed.** All P1/P2 findings from round 1 have been addressed.
